server/src/main/java/org/jboss/as/server/controller/git/ElytronClientSshdSessionFactory.java

```diff
@@ -0,0 +1,226 @@
+            return (new File(sshDir, s)).toPath();
+        }).filter((var0) -> {
+            return Files.exists(var0, new LinkOption[0]);
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public SshdSession getSession(URIish urIish, CredentialsProvider credentialsProvider, FS fs, int tms) throws TransportException {
+        try {
+            this.uri = urIishToUri(urIish);
+            AuthenticationConfiguration config = CLIENT.getAuthenticationConfiguration(this.uri, context);
+            CredentialCallback locationCallback = new CredentialCallback(SSHCredential.class, null);
+            CredentialCallback keyPairCallback = new CredentialCallback(KeyPairCredential.class, null);
+            Callback[] callbacks = {locationCallback, keyPairCallback};
+            CLIENT.getCallbackHandler(config).handle(callbacks);
+
+            if (locationCallback.getCredential() != null) {
+                SSHCredential sshCredential = locationCallback.getCredential().castAs(SSHCredential.class);
+                this.setSshDirectory(sshCredential.getSshDirectory());
+                this.defaultIdentities = sshCredential.getPrivateKeyIdentities();
+                this.knownHostsFile = sshCredential.getKnownHostsFile();
+            }
+
+            if (keyPairCallback.getCredential() != null) {
+                KeyPairCredential keyPairCredential = keyPairCallback.getCredential().castAs(KeyPairCredential.class);
+                this.keyPair = keyPairCredential.getKeyPair();
+            }
+
+        } catch (IOException  | UnsupportedCallbackException | URISyntaxException ex) {
+            ServerLogger.ROOT_LOGGER.failedToLoadSSHCredentials(ex, ex.getMessage());
+        }
+        return super.getSession(urIish, credentialsProvider, fs, tms);
+    }
+
+    @Override
+    protected KeyPasswordProvider createKeyPasswordProvider(CredentialsProvider provider) {
+        URI uri = this.uri;
+       return new IdentityPasswordProvider(new CredentialsProvider() {
+           @Override
+           public boolean isInteractive() {
+               return false;
+           }
+
+           @Override
+           public boolean supports(CredentialItem... credentialItems) {
+               for (CredentialItem i : credentialItems) {
+                   if (!(i instanceof CredentialItem.Password)) {
+                       return false;
+                   }
+               }
+               return true;
+           }
+
+           @Override
+           public boolean get(URIish urIish, CredentialItem... credentialItems) throws UnsupportedCredentialItem {
+               try {
+                   AuthenticationConfiguration config = CLIENT.getAuthenticationConfiguration(uri, context);
+
+                   for (CredentialItem i : credentialItems) {
+                       if (i instanceof CredentialItem.Password) {
+                           CredentialCallback callback = new CredentialCallback(SSHCredential.class, null);
```

Comment:
May be good to add a comment here to indicate why only ```SSHCredential``` is considered here.

Reply:
Fixed, thanks!